File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> store ;
        vector<vector<string>> ans ;
        for(string s : strs){
            string og = s ;
            sort(s.begin(),s.end());
            store[s].push_back(og);
        }
        for (auto &pair : store ) {
            ans.push_back(pair.second);
        }
        return ans ;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        maps = defaultdict(list)
        for x in strs :
            y = "".join(sorted(x))
            maps[y].append(x)
        ans = []
        for x,y in maps.items():
            ans.append(y)
        return ans 

